app/src/main/java/com/duckduckgo/app/browser/BrowserChromeClient.kt

```diff
@@ -73,4 +74,16 @@ class BrowserChromeClient @Inject constructor() : WebChromeClient(), CoroutineSc
         webViewClientListener?.showFileChooser(filePathCallback, fileChooserParams)
         return true
     }
+
+    override fun onCreateWindow(view: WebView?, isDialog: Boolean, isUserGesture: Boolean, resultMsg: Message?): Boolean =
+        if (isUserGesture && resultMsg?.obj is WebView.WebViewTransport) {
+            val transport = resultMsg.obj as WebView.WebViewTransport
+            val newWebView = WebView(view?.context)
+            transport.webView = newWebView
+            resultMsg.sendToTarget()
+            webViewClientListener?.openInNewTab(newWebView.url)
```

Comment:
Rather than creating a webView here that we just throw away (which is a waste of will be losing vital data) we need to associate the `transport` with the WebView that the new tab actually uses. This isn't easy. We need to do something along the lines of `webViewClientListener?.openWebTransportInNewTab(transport)` and then continue to send the transport down to `BrowserTabFragment` via a command type. The fragment will then call the activity `browserActivity?.openWebTransportInNewTab(it.transport)` and the activity will need a new method to handle the creation of a fragment directly (not through live data) and associating the transport with the WebView.

Reply:
Hi @subsymbolic  sure, go for it! I am anyway quite busy these days with my own open-source project [Mockinizer](https://github.com/donfuxx/Mockinizer), so I most likely wouldn't be able to do this anytime soon anyway.